andela/suyabay
# Ft show episodes under various channels

#### What does this PR do?
- Write tests for get channel episodes
- Create controller and a method that shows list of existing podcast episodes in their various channels
#### Description of Task to be completed?
-  Create a controller for channel episodes
-  Create a method that returns channel episodes
-  Use fractal package to transform the channel outputs
-  Write tests that asserts that;
-  A channel name exists and returns a channel with all episodes under it.
-  The channel name supplied by the user does not exist.
-  The channel name exist but it does not have any episode.
#### How should this be manually tested?
- For the test on your terminal run `phpunit tests/tests/ChannelEpisodesTest.php`
- For the main API. you will need POSTMAN, POSTMAN is an extension plugin on google chrome.
- To retrieve all episodes under a channel, set your Request to GET and enter this endpoint  `http://suyabay.app/api/v1/channels/{name}/episodes` if the channel name is correct you should see a JSON data. otherwise you should see;

`{ "data": [] }`

If you wish to customise your search query you can append this query parameters to the above endpoint like `http://suyabay.app/api/v1/channels/{name}/episodes?page=1&limit=10`
#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
#### Screenshots (if appropriate)

![screen shot 2016-05-17 at 6 02 25 pm](https://cloud.githubusercontent.com/assets/14159435/15331521/964efd9c-1c59-11e6-97f2-0a9a261d1d8b.png)

![screen shot 2016-05-17 at 6 03 43 pm](https://cloud.githubusercontent.com/assets/14159435/15331543/b86e2ef2-1c59-11e6-8c7f-f25a6e297608.png)

![screen shot 2016-05-17 at 6 04 08 pm](https://cloud.githubusercontent.com/assets/14159435/15331559/c5588676-1c59-11e6-9d8b-a14f7b66f570.png)
#### Questions:

[@unicodeveloper](https://github.com/unicodeveloper)
[@andela-oadebayo](https://github.com/andela-oadebayo)
